modules/fluid_properties/include/fluidproperties/SinglePhaseFluidProperties.h

```diff
@@ -311,8 +311,20 @@ class SinglePhaseFluidProperties : public FluidProperties
    */
   virtual Real vaporTemperature(Real p) const;
   virtual void vaporTemperature(Real p, Real & Tsat, Real & dTsat_dp) const;
-  DualReal vaporTemperature(const DualReal & p) const;
+  virtual DualReal vaporTemperature(const DualReal & p) const;
+  virtual FPDualReal vaporTemperature_ad(const FPDualReal & p) const;
```

Comment:
`virtual FPDualReal vaporTemperature_ad(const FPDualReal & p) const;`
is there to match the declaration in Water97FluidProperties

https://github.com/idaholab/moose/blob/1145d861a4cf31c9a159563694bd7d5619353a09/modules/fluid_properties/include/fluidproperties/Water97FluidProperties.h#L813

Maybe it's not necessary but I was trying to keep it consistent. There are some inconsistencies between AD versions and non-AD versions throughout Water97 and other fluid properties, some use template functions which I think is the prefered implementation and others use the `_ad` implementation.

Reply:
In Water97:

The general format is that there is a Real version, an ADReal version, and then a void version which takes the derivatives and then another void version which is AD. All of these reference two template functions one with the derivatives and one without.

https://github.com/idaholab/moose/blob/1145d861a4cf31c9a159563694bd7d5619353a09/modules/fluid_properties/src/fluidproperties/Water97FluidProperties.C#L80C1-L107C2

However, for some of the functions (vaporTemperature and T_from_p_h) there's no template function just a `_ad` version which is used for each of the functions. 
https://github.com/idaholab/moose/blob/1145d861a4cf31c9a159563694bd7d5619353a09/modules/fluid_properties/src/fluidproperties/Water97FluidProperties.C#L511C1-L549C2

I think both implementations have the same behavior so I'm not totally sure what the difference is. Also, I'm still not clear on the void versions with the derivatives but they are necessary for the functioning of things like PorousFlowWaterVapor. When I looked into what PFWaterVapor calls it calls `void Water97FluidProperties::vaporTemperature` then  `Water97FluidProperties::vaporTemperature_ad` in 

https://github.com/idaholab/moose/blob/087d66eb2aace3e6b0fe1db7b316749254c301d2/modules/porous_flow/src/userobjects/PorousFlowWaterVapor.C#L105C1-L109C49

Reply:
From memory (it was a long time ago!) those `foo_ad` methods are really just used internally to avoid hand-coding the derivatives. I think the `FPDualReal` ones use a smaller derivative container for speed. None of these are exposed in the public `SinglePhaseFluidProperties` API, so you probably don't want them in TabulatedFluidProperties.

I did have an idea which I'll add in a comment below

Reply:
Removed in d2f638c